source/numerics/data_out_dof_data.cc

```diff
@@ -527,6 +527,45 @@ namespace internal
         return vector[cell_number];
       }
 
+      // FIXME:
+      // explicit instantiations for PETSc complex vectors
+#ifdef PETSC_USE_COMPLEX
+      template <>
```

Comment:
i realised i can avoid those changes altogether by simply not instantiating DataOut for complex PETSc vectors.